src/python/pants/engine/legacy/graph.py

```diff
@@ -89,9 +90,14 @@ def _index(self, roots):
     # Index the ProductGraph.
     for node, state in roots.items():
       if type(state) is not Return:
-        trace = '\n'.join(self._scheduler.trace())
-        raise AddressLookupError(
-            'Build graph construction failed for {}:\n{}'.format(node, trace))
+        # NB: ResolveError means that a target was not found, which is a common user facing error.
+        #     So, we propagate that error's message.
```

Comment:
The message is already propagated below: the reason I think #4446 is related to this one is that we should probably flatten the traces such that the exception message is front and center, rather than deeply indented.

I think maybe the change you've made here is a good stopgap, but would be good to leave a comment about removing the special casing once traces are clearer post #4446.